src/frontend/src/pages/AdminToolsPage/AdminToolsUserManagement.tsx

```diff
@@ -0,0 +1,129 @@
+import Autocomplete from '@mui/material/Autocomplete';
+import { NERButton } from '../../components/NERButton';
+import { Grid } from '@mui/material';
+import PageBlock from '../../layouts/PageBlock';
+import TextField from '@mui/material/TextField';
+import IconButton from '@mui/material/IconButton';
+import SearchIcon from '@mui/icons-material/Search';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import { useState } from 'react';
+import { useAllUsers, useUpdateUserRole } from '../../hooks/users.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { fullNamePipe } from '../../utils/pipes';
+
+const AdminToolsUserMangaement: React.FC = () => {
+  const [role, setRole] = useState('');
+  const [userId, setUserId] = useState(-1);
+  const [isDisabled, setIsDisabled] = useState(true);
+  const [hideSuccessLabel, setHideSuccessLabel] = useState(true);
+  const { isLoading, isError, error, data } = useAllUsers();
+  const update = useUpdateUserRole();
+
+  if (isLoading || !data) return <LoadingIndicator />;
+
+  if (isError) return <ErrorPage message={error?.message} />;
+
+  const getUserRole = () => {
```

Comment:
why not just make the state track the user instead of the userId and if one hasn't been selected it's null